drivers/ws281x/ws281x.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 Marian Buschsieweke
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ws281x
+ *
+ * @{
+ *
+ * @file
+ * @brief       Driver for the WS2812 or the SK6812 RGB LEDs sold as NeoPixel
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+#include <assert.h>
+#include <errno.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "ws281x.h"
+#include "ws281x_constants.h"
+#include "ws281x_params.h"
+#include "periph/gpio.h"
+
+/* Default buffer used in ws281x_params.h. Will be optimized out if unused */
+uint8_t ws281x_buf[WS281X_PARAM_NUMOF * WS281X_BYTES_PER_DEVICE];
+
+int ws281x_init(ws281x_t *dev, const ws281x_params_t *params)
```

Comment:
I just noticed that a backend-specific `ws281x_init()` will be necessary.

Reply:
But I guess this should work for most backends. E.g. the GPIO ABC backend will also use this. And I guess for other slow platforms (e.g. MSP430) manual bitbanging with CPU cycle counting will also be the easiest path. I simple marked the symbol as weak to allow other backends (e.g. the SPI one) to provide their own.